src/umpire/strategy/DynamicPool.cpp

```diff
@@ -180,28 +180,27 @@ void* DynamicPool::allocate(std::size_t bytes)
 
     m_curr_bytes += rounded_bytes;
 
-    const int64_t left_bytes{static_cast<int64_t>(
-        free_size - rounded_bytes)};
+    const std::size_t left_bytes{free_size - rounded_bytes};
 
-    if (left_bytes > m_align_bytes) {
+    if (left_bytes > 0) {
       insertFree(static_cast<unsigned char*>(ptr) + rounded_bytes, left_bytes,
                  false, whole_bytes);
     }
   } else {
     // Allocate new block -- note this does not check whether alignment is met
     const std::size_t alloc_bytes{std::max(rounded_bytes, m_min_alloc_bytes)};
+    std::cout << "Allocating " << alloc_bytes << " from resource for " << rounded_bytes << std::endl;
     ptr = allocateFromResource(alloc_bytes);
 
     // Add used
     insertUsed(ptr, rounded_bytes, true, alloc_bytes);
     m_actual_bytes += alloc_bytes;
     m_curr_bytes += rounded_bytes;
 
-    const int64_t left_bytes{static_cast<int64_t>(
-        m_min_alloc_bytes - rounded_bytes)};
+    const std::size_t left_bytes{alloc_bytes - rounded_bytes};
 
     // Add free
-    if (left_bytes > m_align_bytes)
+    if (left_bytes > 0)
```

Comment:
This was the cause of the issue. The other changes make the code consistent.

Reply:
yikes.  I'm not sure what the intent of the previous line was.

Reply:
I still could see `m_min_alloc_bytes - rounded_bytes` being something huge if `rounded_bytes > m_min_alloc_bytes`.